components/movementsensor/rtkutils/cachedData.go

```diff
@@ -174,3 +178,31 @@ func (g *CachedData) Properties(
 		CompassHeadingSupported: true,
 	}, nil
 }
+
+// calculateCompassDegreeError calculates the compass degree error
+// of two geo points.
+// GPS provides a heading data only when it has a course of direction.
+// This function provides an estimated error for that data.
+func (g *CachedData) calculateCompassDegreeError(p1, p2 *geo.Point) float64 {
+	// Check if the two positions are the same.
+	if p1.Lat() == p2.Lat() && p1.Lng() == p2.Lng() {
+		return math.NaN()
```

Comment:
nope not needed. I'm going replace this logic with nil check. Thanks for pointing this out

Reply:
Susmita and I discussed further: because we're using `atan2` and not merely `atan`, you don't even get divide-by-zero errors when `adjacent` is 0, and we don't need any of these checks! I suspect the thing that's off is that we should swap the two arguments to `atan2` entirely, but I'm not confident of that. Susmita is checking details to be sure.

Reply:
Susmita checked it out: the arguments are in the right order. When `adjacent` is either 0 or even close to 0, the error is supposed to be pi/2 (huge), whereas when `adjacent` is large, the error should be small. That's what we've got as-is.